src/vcCamera.cpp

```diff
@@ -185,7 +185,12 @@ void vcCamera_Apply(vcState *pProgramState, vcCamera *pCamera, vcCameraSettings
     if (isnan(pCamera->position.x) || isnan(pCamera->position.y) || isnan(pCamera->position.z))
       pCamera->position = udDouble3::zero();
 
-    pCamInput->smoothRotation += udDouble2::create(-pCamInput->mouseInput.x, pCamInput->mouseInput.y) * 0.5; // Because this messes with HEADING, has to be reversed
+    udDouble2 rotation = udDouble2::create(-pCamInput->mouseInput.x, pCamInput->mouseInput.y) * 0.5; // Because this messes with HEADING, has to be reversed
+    udDouble2 result = pCamera->headingPitch + rotation;
+    if (result.y > UD_HALF_PI)
+      rotation.y = UD_HALF_PI - pCamera->headingPitch.y;
```

Comment:
Does this stop going past -UD_HALF_PI as well? (looking straight down)

Reply:
Fixed.